src/NHibernate/Context/WcfOperationSessionContext.cs

```diff
@@ -50,3 +56,36 @@ public class WcfStateExtension : IExtension<OperationContext>
 		public void Detach(OperationContext owner) { }
 	}
 }
+#else
+// 6.0 TODO: remove the whole #else
+using System;
+using System.Collections;
+using NHibernate.Engine;
+
+namespace NHibernate.Context
+{
+	/// <summary>
+	/// Obsolete class not usable in the .Net Core and .Net Standard distributions of NHibernate. Use the
+	/// .Net Framework distribution of NHibernate if you need it. See
+	/// https://github.com/nhibernate/nhibernate-core/issues/1842
+	/// </summary>
+	[Obsolete("Not supported in the .Net Core and .Net Standard distributions of NHibernate")]
+	public class WcfOperationSessionContext : MapBasedSessionContext
+	{
+		public WcfOperationSessionContext(ISessionFactoryImplementor factory) : base(factory)
+		{
+			throw new NotSupportedException(
+				"WcfOperationSessionContext is currently supported only by the .Net Framework distribution of NHibernate");
+		}
+
+		protected override IDictionary GetMap()
+		{
+			return null;
+		}
+
+		protected override void SetMap(IDictionary value)
+		{
+		}
+	}
+}
```

Comment:
The `#else` part defining a dummy `WcfOperationSessionContext` is unfortunately not enough to avoid a binary breaking change for .Net Core and .Net Standard: there is also the `WcfStateExtension` class which was unfortunately public. But it cannot be put back in that `#else`, because it implements the `IExtension<OperationContext>` interface from `System.ServiceModel`.

May this change still be acceptable for a 5.2? Or is it anyway a no-go, even if no one should have referenced that class, especially in a .Net Core application?

Reply:
It should not, but I'm ok with that.